src/components/data/TableCell.tsx

```diff
@@ -8,7 +8,10 @@ import TableContext from './TableContext';
 import TableSectionContext from './TableSectionContext';
 
 export type TableCellProps = PresentationalProps &
-  Omit<JSX.HTMLAttributes<HTMLElement>, 'size'>;
+  Omit<JSX.HTMLAttributes<HTMLElement>, 'size'> & {
+    /** Do not apply a default padding, allowing consuming code to control it */
```

Comment:
```suggestion
    /** Remove default padding, allowing consuming code to control it */
```

I think this makes it a bit easier to reason about what `true` and `false` values mean ("Do/don't remove default padding" vs "Do/don't not apply a default padding").